File: leetcode-python-go/1732. 找到最高海拔/main.go
package main

import (
	"fmt"
)

func largestAltitude(gain []int) (ans int) {
	total := 0
	for _, x := range gain {
		total += x
		ans = max(ans, total)
	}
	return
}

func max(a, b int) int {
	if b > a {
		return b
	}
	return a
}

func main() {
	gain := []int{-5, 1, 5, 0, -7}
	results := largestAltitude(gain)
	fmt.Println(results)
}

File: leetcode-python-go/1732. 找到最高海拔/main.py
from typing import List

gain = [-5,1,5,0,-7]

def largestAltitude(gain: List[int]) -> int:
    ans = total = 0
    for x in gain:
        total += x
        ans = max(ans, total)
    return ans


print(largestAltitude(gain))